var/spack/repos/builtin/packages/libfabric/package.py

```diff
@@ -44,8 +45,10 @@ class Libfabric(AutotoolsPackage):
                'gni',
                'xpmem',
                'udp',
+               'tcp',
                'rxm',
-               'rxd')
+               'rxd',
+               'shm')
```

Comment:
Are these new fabrics available for all versions of `libfabric`?

Reply:
Ah, good catch. Yeah, these are newer providers that were not available with earlier versions. What's the best way to handle this with spack?

Reply:
Hmm, I don't know of a great way to handle this yet. You can add `conflicts()` statements so that someone trying to request a fabric for a version that doesn't support it gets an informative error message. See the `gcc` package for examples.

Reply:
Alright, will take a look at a couple other packages to see how this is handled. Maybe what I'll do is just get the 1.6 update in first so some of our internal workflows can proceed, and then will look at adding in the newer providers and the bugfix versions.